playground/src/client/index.ts

```diff
@@ -53,9 +52,48 @@ async function measureLcp({
   await page.waitForNavigation({
     waitUntil: 'networkidle0',
   });
-  const lcpResult = await page.evaluate(getObserverResult);
-  await page.close();
-  return lcpResult;
+  return await page.evaluate(getObserverResult);
+}
+
+// The method to isolate cache between multiple tests.
+enum IsolationMode {
+  // The entire browser is cleared before running a test. The drawback is that
+  // tests can not run in parallel.
+  ClearBrowserCache,
+  // Each test runs in an individual incognito browser context. The drawback is
+  // that incognito mode might behave differently from regular context.
+  IncognitoBrowserContext,
+}
+
+// Opens a new Chrome tab (page), and measures the LCP of given URL.
+async function createPageAndMeasureLcp({
+  browser,
+  isolationMode,
+  url,
+  useSxg,
+}: {
+  browser: Browser;
+  isolationMode: IsolationMode;
+  url: string;
+  useSxg: boolean;
+}) {
+  if (isolationMode === IsolationMode.IncognitoBrowserContext) {
+    const context = await browser.createIncognitoBrowserContext();
```

Comment:
Possibly good news: It looks like this uses [`Target.createBrowserContext`](https://github.com/puppeteer/puppeteer/blob/7748730163bc1a14cbb30881809ea529844f887e/src/common/Browser.ts#L324-L330) which is documented as [`Similar to an incognito profile`](https://chromedevtools.github.io/devtools-protocol/tot/Target/#method-createBrowserContext). Since "similar" != "same as"... maybe this doesn't block third-party cookies?

Reply:
`CreateBrowserContext` [does set](https://source.chromium.org/chromium/chromium/src/+/main:headless/lib/browser/headless_devtools_manager_delegate.cc;l=89;drc=2e194e180046e5583bad6a226fa942c95e5bad52) it into incognito mode, and the Chrome UI shows an incognito icon on the top right. I have tested that at least [this function](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/estimate) returns a different value from the default browser context. But I haven't find any tests for the third-party cookie.

Reply:
[Here is a test](http://outer.twifkak.com/cookies2/outer.html) that will display a date if third-party cookies are allowed, and a blank frame otherwise.